src/detect-ssh-hassh-server.c

```diff
@@ -0,0 +1,209 @@
+    for ( ; sm != NULL; sm = sm->next)
+    {
+        if (sm->type != DETECT_CONTENT)
+            continue;
+
+        const DetectContentData *cd = (DetectContentData *)sm->ctx;
+
+        if (cd->flags & DETECT_CONTENT_NOCASE) {
+            *sigerror = "ssh.hassh.server should not be used together with "
+                        "nocase, since the rule is automatically "
+                        "lowercased anyway which makes nocase redundant.";
+            SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+        }
+
+        if (cd->content_len != 32)
+        {
+            *sigerror = "Invalid length of the specified ssh.hassh.server (should "
+                        "be 32 characters long). This rule will therefore "
+                        "never match.";
+            SCLogWarning(SC_WARN_POOR_RULE,  "rule %u: %s", s->id, *sigerror);
+            return FALSE;
+        }
+        for (size_t i = 0; i < cd->content_len; ++i)
+        {
+            if(!isxdigit(cd->content[i])) 
+            {
+                *sigerror = "Invalid ssh.hassh.server string (should be string of hexademical characters)."
+                            "This rule will therefore never match.";
+                SCLogWarning(SC_WARN_POOR_RULE,  "rule %u: %s", s->id, *sigerror);
+                return FALSE;
+            }
+        }
+    }
+
+    return TRUE;
+}
+
+static void DetectSshHasshServerHashSetupCallback(const DetectEngineCtx *de_ctx,
+                                          Signature *s)
+{
+    SigMatch *sm = s->init_data->smlists[g_ssh_hassh_buffer_id];
+    for ( ; sm != NULL; sm = sm->next)
+    {
+        if (sm->type != DETECT_CONTENT)
+            continue;
+
+        DetectContentData *cd = (DetectContentData *)sm->ctx;
+
+        uint32_t u;
+        for (u = 0; u < cd->content_len; u++)
+        {
+            if (isupper(cd->content[u])) {
+                cd->content[u] = tolower(cd->content[u]);
+            }
+        }
+
+        SpmDestroyCtx(cd->spm_ctx);
+        cd->spm_ctx = SpmInitCtx(cd->content, cd->content_len, 1,
+        		de_ctx->spm_global_thread_ctx);
+    }
+}
+
+/**
+ * \brief Registration function for hasshServer keyword.
+ */
+void DetectSshHasshServerRegister(void) 
+{
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].name = KEYWORD_NAME;
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].alias = KEYWORD_ALIAS;
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].desc = BUFFER_NAME " sticky buffer";
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].RegisterTests = NULL;
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].url = "/rules/" KEYWORD_DOC;
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].Setup = DetectSshHasshServerSetup;
+    sigmatch_table[DETECT_AL_SSH_HASSH_SERVER].flags |= SIGMATCH_INFO_STICKY_BUFFER | SIGMATCH_NOOPT;
+
+    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOCLIENT, 2, PrefilterGenericMpmRegister, GetSshData, ALPROTO_SSH, SshStateBannerDone),
```

Comment:
style: lines too long. Can you check the other places as well?

Reply:
yes, fixed all detect-ssh-hassh-** files